src/js/player.js

```diff
@@ -650,22 +649,25 @@ class Player extends Component {
     this.isReady_ = false;
 
     // Grab tech-specific options from player options and add source and parent element to use.
-    const techOptions = assign({
+    let techOptions = {
       source,
       'nativeControlsForTouch': this.options_.nativeControlsForTouch,
       'playerId': this.id(),
       'techId': `${this.id()}_${techName}_api`,
-      'videoTracks': this.videoTracks_,
-      'textTracks': this.textTracks_,
-      'audioTracks': this.audioTracks_,
       'autoplay': this.options_.autoplay,
       'preload': this.options_.preload,
       'loop': this.options_.loop,
       'muted': this.options_.muted,
       'poster': this.poster(),
       'language': this.language(),
       'vtt.js': this.options_['vtt.js']
-    }, this.options_[techName.toLowerCase()]);
+    };
+
+    Object.keys(TRACK_TYPES).forEach((type) => {
```

Comment:
I wonder if this array ought to be stored somewhere instead of calling `Object.keys(TRACK_TYPES)` in multiple places. Either exported from `tracks/track-types` or as a private variable in this module.